gateway/mw_ip_blacklist.go

```diff
@@ -18,7 +18,12 @@ func (i *IPBlackListMiddleware) Name() string {
 }
 
 func (i *IPBlackListMiddleware) EnabledForSpec() bool {
-	return i.Spec.EnableIpBlacklisting && len(i.Spec.BlacklistedIPs) > 0
+	enabled := i.Spec.APIDefinition.EnableIpBlacklisting || !i.Spec.APIDefinition.IPAccessControlDisabled
+	if !enabled {
+		return false
+	}
+
+	return len(i.Spec.APIDefinition.BlacklistedIPs) > 0
```

Comment:
```suggestion
	if i.Spec.APIDefinition.IPAccessControlDisabled {
		return false
	}

	return i.Spec.EnableIpBlacklisting && len(i.Spec.BlacklistedIPs) > 0
```

for clarity, smaller change

Reply:
reverted this. 
it expects `i.Spec.EnableIpBlacklisting` to be true, but it need not be true when `IPAccessControlDisabled` is false.